contracts/utils/contracts/src/LibAddressArrayRichErrors.sol

```diff
@@ -0,0 +1,46 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "./LibRichErrors.sol";
+
+
+library LibAddressArrayRichErrors {
+
+    using LibRichErrors for *;
+
+    // bytes4(keccak256("MismanagedMemoryError(uint256,uint256)"))
+    bytes4 internal constant MISMANAGED_MEMORY_ERROR_SELECTOR =
+        0x5fc83722;
+
+    // solhint-disable func-name-mixedcase
+    function MismanagedMemoryErrorRevert(
+        uint256 freeMemPtr,
+        uint256 addressArrayEndPtr
+    )
+        internal
+        pure
+    {
+        abi.encodeWithSelector(
+            MISMANAGED_MEMORY_ERROR_SELECTOR,
+            freeMemPtr,
+            addressArrayEndPtr
+        )._rrevert();
```

Comment:
I don't mind this style of reverting within the error function itself, but why are we using a pattern that is different from all of our other rich reverts? I prefer to be consistent, and it's probably easier to change this than it is to change all of our other rich reverts.

Reply:
The reason for creating the new pattern was to create a nice and reusable pattern for implementing RichErrors in _libraries_. As far as I know (I've greeped in most of the contracts directories for "_rrevert"), we do not use RichErrors in any libraries at the moment, so there wasn't an established style that I saw. 

Now, the reason that we need a new style for libraries is because they cannot inherit from other contracts. This is the pattern that we usually use to get RichErrors, so I needed to use a different mechanism, which was the `using ... for ...` pattern of libraries. I actually think that this is clean enough to consider making it the standard for RichErrors because the library pattern will work on either contracts or libraries.

All of this said, if you know of a way to avoid making a new pattern, I'll happily adopt that